src/components/infinite-scroll/InfiniteScroll.tsx

```diff
@@ -0,0 +1,108 @@
+/**
+ * From https://github.com/jaredpalmer/react-simple-infinite-scroll
+ * Updated to accept a scroll container React ref as the "window"
+ */
+import * as React from 'react';
+import throttleFn from 'lodash/throttle';
+
+export type Props = {
+    /** Components to render, should include entities/loading states/end message */
+    children: React.ReactNode;
+    /** Does the resource have more entities */
+    hasMore: boolean;
+    /** Are more entities being loaded */
+    isLoading: boolean;
+    /** Callback to load more entities */
+    onLoadMore: () => void;
+    /** React ref of the scroll container.
+     * Used to listen to scroll events and calculate onLoadMore calls.
+     * Set useWindow to true to use the window as scroll container.
+     */
+    scrollContainerNode?: HTMLElement;
+    /** Scroll threshold */
+    threshold?: number;
+    /** Throttle rate */
+    throttle?: number;
+    /** Set useWindow to true to use the window as scroll container. If set to true, will ignore scrollContainerRef. */
+    useWindow?: boolean;
+};
+
+function InfiniteScroll({
```

Comment:
This is definitely a preference thing, but I like to use React.FC and pass props as a generic (and I also like arrow functions):
`const InfiniteScroll: React.FC<InfiniteScrollProps> = ({kitty, turtle}) => <div>kittle is {kitty + turtle} </div>`
From my perspective, would be cool if we ended up standardized using React.FC<Props> in the migration docs 🙏

Reply:
@jcamden what are the benefits of React.FC? I know there is some controversy surrounding it...

Reply:
Sorry, I disagree completely here and I would ban using React.FC altogether.  I saw milion times people using it by default even when their components were not using `children`  (FC adds children by default). 
So I think it's always better to be explicit and type children per component basis. 
As for arrow function - I am not a fan unless it's lambda or you need to keep `this` value. Not sure why people prefer it over `function` - for my eyes they're less readable.

So unless you feel strongly about it I would prefer to keep `function` keyword

Reply:
BTW: all those React.FC that you have in the code will be a problem when moving to @types/react 18 since they removed the `children` :)

Reply:
Yep, you're totally right. I just learned another thing.

Reply:
As a follow up, if we're not going to use React.FC, would it be a good idea for us to type the return value (i.e. React.ReactNode) instead of relying on inference? (Btw, thanks for helping us discuss Typescript with this PR :) )